client/src/task.rs

```diff
@@ -92,3 +89,50 @@ pub enum Histogram {
         chunk_length: Option<u64>,
     },
 }
+
+impl Histogram {
+    /// The length of this histogram, i.e. the number of buckets.
+    pub fn length(&self) -> usize {
+        match self {
+            Histogram::Categorical { buckets, .. } => buckets.len(),
+            Histogram::Continuous { buckets, .. } => buckets.len() + 1,
+            Histogram::Length { length, .. } => *length as usize,
+        }
+    }
+
+    /// The chunk length used in the VDAF.
+    pub fn chunk_length(&self) -> usize {
+        match self {
+            Histogram::Categorical { chunk_length, .. } => chunk_length,
+            Histogram::Continuous { chunk_length, .. } => chunk_length,
+            Histogram::Length { chunk_length, .. } => chunk_length,
+        }
+        .map(|c| c as usize)
+        .unwrap_or_else(|| optimal_chunk_length(self.length()))
```

Comment:
AIUI chunk_length being absent is a peculiarity only of DAP-04, which doesn't have a chunk length parameter.

Tough to say what we _should_ do here, removing DAP-04 logic can't come soon enough.